examples_tests/31.SSBOBenchmark/main.cpp

```diff
@@ -0,0 +1,401 @@
+
+    IAssetLoader::SAssetLoadParams lp;
+
+    auto vertexShaderBundle = am->getAsset("../ssboBenchmarkShaders/vertexShader.vert", lp);
+    auto fragmentShaderBundle = am->getAsset("../ssboBenchmarkShaders/fragmentShader.frag", lp);
+
+    CShaderIntrospector introspector(am->getGLSLCompiler());
+    const auto extensions = driver->getSupportedGLSLExtensions();
+
+    //create cpu shaders
+    ICPUSpecializedShader* shaders[2] =
+    {
+        IAsset::castDown<ICPUSpecializedShader>(vertexShaderBundle.getContents().first->get()),
+        IAsset::castDown<ICPUSpecializedShader>(fragmentShaderBundle.getContents().first->get())
+    };
+    auto cpuPipeline = introspector.createApproximateRenderpassIndependentPipelineFromIntrospection(shaders, shaders + 2, extensions->begin(), extensions->end());
+
+    //temporary
+    constexpr size_t diskCount = 3001;
+
+    std::vector<uint16_t> tesselation(diskCount);
+
+    //get random tesselation for disks
+    {
+        std::random_device rd;
+        std::mt19937 mt(rd());
+        std::uniform_int_distribution<uint32_t> dist(15, 14999);
+
+        //TODO: test
+        //`dist(mt) | 0x0001` so vertexCount is always odd (only when diskCount is odd as well)
+        std::generate(tesselation.begin(), tesselation.end(), [&]() { return dist(mt) | 0x0001; });
+    }
+
+    //set mesh params
+    const size_t vertexCount = std::accumulate<core::vector<uint16_t>::iterator, size_t>(tesselation.begin(), tesselation.end(), 0) + 2 * diskCount; //sum + 2 * diskCount because vertex count of each disk is equal to: (tesselation + 2)
+    const size_t indexCount  = std::accumulate<core::vector<uint16_t>::iterator, size_t>(tesselation.begin(), tesselation.end(), 0) * 3;
+    E_INDEX_TYPE indexType = E_INDEX_TYPE::EIT_32BIT;
+    constexpr uint32_t diskVertexSize = 30;
+    constexpr uint32_t vertexSize = sizeof(Vertex);
+    SBufferBinding<ICPUBuffer> vertexBuffer{ 0, core::make_smart_refctd_ptr<ICPUBuffer>(vertexCount * vertexSize) };
+    // SBufferBinding<ICPUBuffer> bindings[ICPUMeshBuffer::MAX_ATTR_BUF_BINDING_COUNT];
+    SBufferBinding<ICPUBuffer> indexBuffer{ 0, core::make_smart_refctd_ptr<ICPUBuffer>(indexCount * sizeof(uint32_t)) };
+
+    SPrimitiveAssemblyParams assemblyParams;
+    assemblyParams.primitiveType = E_PRIMITIVE_TOPOLOGY::EPT_TRIANGLE_LIST;
+
+    SVertexInputParams vertexInputParams = 
+    { 
+        0b1111u, 0b1u,
+        {
+            {0u,EF_R32_UINT,offsetof(Vertex,boneID)},
+            {0u,EF_R32G32B32_SFLOAT,offsetof(Vertex,pos)},
+            {0u,EF_R8G8B8A8_UNORM,offsetof(Vertex,color)},
+            {0u,EF_R8G8_USCALED,offsetof(Vertex,uv)},
+            {0u,EF_R32G32B32_SFLOAT,offsetof(Vertex,normal)}
+        },
+        {vertexSize,EVIR_PER_VERTEX} 
+    };
+
+
+    core::vector<core::matrix3x4SIMD> SSBOData; //bone matrices will also reside here
+    SSBOData.reserve(diskCount);
+
+    //create disks and join them into one mesh buffer
```

Comment:
this I want you to separate into a utility class like we talked with @AnastaZIuk and @Crisspl